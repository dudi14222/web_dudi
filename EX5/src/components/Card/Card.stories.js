import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import {
  withKnobs
  
} from '@kadira/storybook-addon-knobs';

import Card from './'


const storiesCard = storiesOf('Card', module);
storiesCard.addDecorator(withKnobs);
storiesCard.addWithInfo('with children', '', () => {
  const story = <Card>
    <div>some text</div>
  </Card>;
  return story;
});

storiesCard.addWithInfo('with some props', '', () => {
  const story = <Card style={{ display: 'flex', flexWrap: 'wrap', justifyContent: 'center' }}
    onClick={action('clicked')}
  >
    Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.
  </Card>;
  return story;
});
