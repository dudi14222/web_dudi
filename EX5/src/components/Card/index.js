import React from 'react';
import './card.css';
import classNames from 'classnames';

const Card = ({ children, className: parentClassName, ...props }) => (
    <div className={classNames(['raised-2','card', parentClassName])} {...props}>
      { children }
    </div>
);
export default Card;