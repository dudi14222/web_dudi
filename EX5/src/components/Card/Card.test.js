import Card from '../Card';
import React from 'react';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';
import { shallow } from "enzyme";



describe('Card snapshot tests', () => {
    it('test snapshot', () => {
        const tree = renderer.create(
            <MemoryRouter>
                <Card />
            </MemoryRouter>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
})

describe('Card shallow tests', () => {
    it('should render children when passed in', () => {
        const wrapper = shallow((
            <Card>
                <div className="unique" />
            </Card>
        ));    
        expect(wrapper.contains(<div className="unique" />)).toBe(true);
    });
})



