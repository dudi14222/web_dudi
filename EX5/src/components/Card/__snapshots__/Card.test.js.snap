// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card snapshot tests test snapshot 1`] = `
<div
  className="raised-2 card"
/>
`;
